blockbin/src/components/InputForm.js

```diff
@@ -1,16 +1,73 @@
 import React, { Component } from 'react';
+import Web3 from 'web3';
 
 import '../App.css';
 
+const CONTRACT_ADDRESS = '0xa022cc6ff8895fd07b03c480d46f20c412716fdc';
+
 class InputForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      pastedText: 'What would you like to put on the ~blockchain~?',
+    };
+
+    if (typeof window.web3 !== 'undefined') {
+      console.log('using mist or metamask');
+      const newWeb3 = new Web3(window.web3.currentProvider);
+      this.web3 = newWeb3;
+    } else {
+      console.log('why aren\'t you using mist or metamask?!');
+      // set the provider you want from Web3.providers
+      this.web3 = new Web3(new Web3.providers.HttpProvider("http://localhost:8545"));
+    }
+    // Replace with correct ABI
```

Comment:
This should probably live in a separate file? Feels weird to have this in the InputForm component :)

(but no big deal, we can split later too)